.github/workflows/pullrequest.yml

```diff
@@ -56,3 +56,35 @@ jobs:
       - run: cargo fmt --all -- --check
       - run: cargo doc --all --exclude plexi-cli --all-features --document-private-items
       - run: cargo test --all --verbose --exclude plexi-cli --all-features
+
+  deploy-rust:
+    name: Deploy Rust Crates
+    timeout-minutes: 5
+    strategy:
+      matrix:
+        os: [ubuntu-24.04]
+    runs-on: ${{ matrix.os }}
+    if: ${{ github.ref == 'refs/heads/main' }}
```

Comment:
do we really want to publish when on main? Having a release model could be better, like we do withb privacypass https://github.com/cloudflare/privacypass-ts/blob/main/.github/workflows/release.yml#L3-L6

Reply:
The difficulty I'm encountering with releases is that the `npm` and the Rust crates need to be able to go out separately, but afaik Github releases use Git tags, which couple the two inextricably. I don't have a better model in hand right now than defaulting to pushing on `main` and failing - any suggestions?

Reply:
there's possibilities to filter based on the release tag name, which would be better. I understand this creates a more complex configuration though. Having a ticket to track "proper release" seems to be a minimum to call this closed.